Content.Server/Explosions/ExplosionHelper.cs

```diff
@@ -26,132 +28,126 @@ public static class ExplosionHelper
         /// </summary>
         private static readonly Vector2 EpicenterDistance = (0.1f, 0.1f);
 
-        public static void SpawnExplosion(this EntityCoordinates coords, int devastationRange, int heavyImpactRange, int lightImpactRange, int flashRange)
+        private static ExplosionSeverity CalculateSeverity(in float distance, in float devastationRange, in float heaveyRange)
         {
-            var tileDefinitionManager = IoCManager.Resolve<ITileDefinitionManager>();
+            if (distance < devastationRange)
+            {
+                return ExplosionSeverity.Destruction;
+            }
+            else if (distance < heaveyRange)
+            {
+                return ExplosionSeverity.Heavy;
+            }
+            else
+            {
+                return ExplosionSeverity.Light;
+            }
+        }
+
+        private static void DamageEntitiesAndTiles(this EntityCoordinates epicenter, in float devastationRange, in float heaveyRange, in float maxrange)
+        {
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+            var mapManager = IoCManager.Resolve<IMapManager>();
+            if (!mapManager.TryGetGrid(epicenter.GetGridId(entityManager), out var mapGrid))
+            {
+                return;
+            }
+
             var serverEntityManager = IoCManager.Resolve<IServerEntityManager>();
             var entitySystemManager = IoCManager.Resolve<IEntitySystemManager>();
-            var mapManager = IoCManager.Resolve<IMapManager>();
+            var exAct = entitySystemManager.GetEntitySystem<ActSystem>();
+            var tileDefinitionManager = IoCManager.Resolve<ITileDefinitionManager>();
             var robustRandom = IoCManager.Resolve<IRobustRandom>();
-            var entityManager = IoCManager.Resolve<IEntityManager>();
 
-            var maxRange = MathHelper.Max(devastationRange, heavyImpactRange, lightImpactRange, 0f);
-            //Entity damage calculation
-            var entitiesAll = serverEntityManager.GetEntitiesInRange(coords, maxRange).ToList();
-
-            foreach (var entity in entitiesAll)
+            var alive_entities_airtight = new HashSet<Vector2i>();
+            var entities_in = serverEntityManager.GetEntitiesInRange(epicenter, maxrange * 2).ToList();
```

Comment:
GetEntitiesInRange makes a Box2 of lengths [range, range]. So you are only getting entities at half the range from the center.

Reply:
HUH that makes me wonder if the implementation is wrong, I'll check on that. I think you should leave it without the multiplication for now since that's what all the other usages of that method do.

Reply:
created this: https://github.com/space-wizards/RobustToolbox/issues/1465